### PR TITLE
tpcc: add thinking/keying time and cap workers

### PR DESCRIPTION
**Pull request by jordanlewis**
This commit corrects the tpcc implementation to be to spec with regards
to concurrency and tpmC.

Specifically, all workers now have a keying/thinking time sleep
before/after the actual operation. These sleeps depend on the operation
being performed and are coded to the tpcc spec.

Also, each warehouse now has a maximum of 10 workers as dictated by the
spec. This means that the tpmC of a tpcc load test should be between 8
and 12 tpmC per warehouse, which requires any interesting test with tpcc
to have a lot more data than we've been testing with.

You can disable this new functionality with the -no-wait flag, which
removes the thinking/keying time and cap on concurrency.

### CODE REVIEW

File: tpcc/worker.go [hunk 1]
```diff
@@ -166,5 +197,11 @@ func (w *worker) run(errCh chan<- error, wg *sync.WaitGroup) {
 		if *maxOps > 0 && v >= *maxOps {
 			return
 		}
+
+		thinkTime := -math.Log(rand.Float64()) * float64(t.thinkTime)
+		if thinkTime > (t.thinkTime * 10) {
+			thinkTime = t.thinkTime * 10
```
Comment (petermattis) [inline tpcc/worker.go]: Where is the `* 10` coming from?

**Review by petermattis — APPROVED** [review]
LGTM

File: tpcc/worker.go (hunk 1 above)
Comment (jordanlewis) [reply tpcc/worker.go]: It's part of the spec. I will add a comment. The think time distribution "may be truncated at 10 times its mean value".

File: tpcc/worker.go [hunk 2]
```diff
@@ -148,7 +172,14 @@ func (w *worker) run(errCh chan<- error, wg *sync.WaitGroup) {
 				break
 			}
 		}
-		if _, err := t.run(w.db, rand.Intn(*warehouses)); err != nil {
+		if *noWait {
+			warehouseID = rand.Intn(*warehouses)
+		}
+
+		time.Sleep(time.Duration(t.keyingTime) * time.Second)
```
Comment (rjnn) [inline tpcc/worker.go]: shouldn't we skip the `time.Sleep` if we're in the `*noWait` case?

**Review by rjnn — APPROVED** [review]
LGTM

File: tpcc/worker.go (hunk 2 above)
Comment (jordanlewis) [reply tpcc/worker.go]: Yes. Thanks for catching that. Done

**Pull request merged**